### PR TITLE
feat(cli): implement runtime rule execution

### PR DESCRIPTION
**Pull request by thecodedrift**
Implements the `runtime-rule-execution` OpenSpec change — the local harness that evaluates a runtime rule in `taskless check`.

**Stacked on #49** (`jakob/runtime-rule-execution`, the proposal) → base this against that branch, not `main`.

All task groups complete (per the repo's `/opsx:apply` convention — commit + pause between groups):

- [x] **1. Runtime-rule recognition** — `discover.ts` + contract types (`types/runtime-rule.ts`)
- [x] **2. Narrow → gate → check harness** — `narrow.ts` (one ast-grep scan/mode), `invoke.ts` (check.ts via `tsx`, timeout + throw isolation), `harness.ts` (`Finding`→`CheckResult`)
- [x] **3. tsx bundling** — CLI dep, runtime-resolved, externalized from the Vite bundle
- [x] **4. Reconcile scoping & materialization** — `runtime/run-set.ts`: sign **check.ts only**, per-rule eligibility, materialize blessed into `.run/`
- [x] **5. check dispatch & modes** — `check.ts` rewrite: static always runs, runtime gated by the mode table, `--dangerously-run-scripts` + `--timeout`, `--json` `skipped`. **Cutover**: removed static-reconcile gating (deleted `run-set.ts`).
- [x] **6. Help & docs** — `check.txt` (topic v2) + `ci.txt`
- [x] **7. Tests & verification** — `runtime-harness.test.ts` + `runtime-check.test.ts`

**Model**: static ast-grep rules are trusted/inert and always run; a runtime rule's `check.ts` is arbitrary code and runs only when server-verified (authed reconcile) or under `--dangerously-run-scripts`. Only the `check.ts` is signed/reconciled.

`typecheck` + `lint` + full `test` green (**338 tests**). Also fixed two real bugs found while testing: a `Finding`→`CheckResult` off-by-one, and a timeout that killed only the `tsx` wrapper (now SIGKILLs the process group).

Refs TSKL-245

**Stack** (root → tip):

- #47
  - #49
    - ➡️ #50 (you are here)

### CODE REVIEW

File: packages/cli/src/rules/runtime/run-set.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+/** Map signed runtime rules to the reconcile report (`check.ts` path + signature). */
+export function reportRuntimeChecks(
+  cwd: string,
+  signed: SignedRuntimeRule[]
+): ReportedFile[] {
+  return signed.map(({ rule, signature }) => ({
+    file: relative(cwd, rule.checkFile),
+    signature,
+  }));
```
[the remark this answers is not in the document]
Comment (thecodedrift) [reply packages/cli/src/rules/runtime/run-set.ts:59]: Fixed — `reportRuntimeChecks` now POSIX-normalizes the reported path (`split(sep).join('/')`) so Windows backslashes can't defeat the server-side path match.

*— AI Coding Agent*

File: packages/cli/src/rules/runtime/narrow.ts [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+      });
+    }
+
+    if (broad.length > 0) {
+      const config = await writeRuleConfig(join(workDirectory, "broad"), broad);
+      // A single ruleId can't be recovered from --files-with-matches; attribute
+      // broad matches to the (usually sole) broad capture rule.
+      const broadRule = broad[0]!;
+      const seen = new Set<string>();
+      await runSg(root, config, ["--files-with-matches"], paths, (line) => {
```
[the remark this answers is not in the document]
Comment (thecodedrift) [reply packages/cli/src/rules/runtime/narrow.ts]: Fixed — the narrow now runs one `--files-with-matches` scan per broad capture rule and attributes each match to that rule, instead of labeling all broad matches as the first rule.

*— AI Coding Agent*

File: packages/cli/src/rules/runtime/invoke.ts [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+      );
+      const stderrChunks: string[] = [];
+      let timedOut = false;
+      const killTree = () => {
+        try {
+          if (child.pid !== undefined) process.kill(-child.pid, "SIGKILL");
+        } catch {
+          child.kill("SIGKILL");
+        }
+      };
```
[the remark this answers is not in the document]
Comment (thecodedrift) [reply packages/cli/src/rules/runtime/invoke.ts:121]: Fixed — on Windows the timeout now kills the tree via `taskkill /T /F` (negative PIDs aren't supported there), so a runaway check is actually terminated.

*— AI Coding Agent*

File: packages/cli/src/rules/runtime/discover.ts [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+    // Every capture rule of a runtime rule names the same `check.ts`; the
+    // generator always writes `check.ts`, so fall back to that.
+    const checkName =
+      captureRules[0]!.rule.metadata.taskless.check || "check.ts";
+    rules.push({
+      name: entry.name,
+      dir: directory,
+      captureRules,
+      checkFile: join(directory, checkName),
+    });
```
[the remark this answers is not in the document]
Comment (thecodedrift) [reply packages/cli/src/rules/runtime/discover.ts]: Fixed — the check file is pinned to `check.ts` inside the rule directory; `metadata.taskless.check` is no longer resolved as a path, so nothing can escape the dir via `../`.

*— AI Coding Agent*

File: packages/cli/src/rules/runtime/narrow.ts [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+          new Error(
+            `ast-grep narrow failed (exit ${String(code)})${
+              stderrChunks.length > 0 ? `: ${stderrChunks.join("").trim()}` : ""
+            }`
+          )
+        );
+        return;
+      }
+      resolve();
+    });
```
[the remark this answers is not in the document]
Comment (thecodedrift) [reply packages/cli/src/rules/runtime/narrow.ts]: Fixed — a `null` exit code (signal termination) is now treated as a narrow failure (with the signal name) instead of silently succeeding.

*— AI Coding Agent*

File: packages/cli/src/rules/runtime/narrow.ts [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+        });
+      });
+    }
+
+    if (broad.length > 0) {
+      const config = await writeRuleConfig(join(workDirectory, "broad"), broad);
+      // A single ruleId can't be recovered from --files-with-matches; attribute
+      // broad matches to the (usually sole) broad capture rule.
+      const broadRule = broad[0]!;
+      const seen = new Set<string>();
```
[the remark this answers is not in the document]
Comment (thecodedrift) [reply packages/cli/src/rules/runtime/narrow.ts]: Fixed — resolved by running one broad scan per broad capture rule so each match is attributed to the correct rule.

*— AI Coding Agent*

File: packages/cli/src/commands/check.ts [hunk 7]
```diff
@@ -95,79 +101,121 @@ function extractPositionalPaths(rawArguments: string[]): string[] {
...
+    );
   }
 
-  const signed = await signRuleFiles(cwd, ruleFiles);
+  const signed = await signRuntimeChecks(discovered);
   const outcome = await reconcile(token, {
     repositoryUrl,
-    files: signed.map(({ file, signature }) => ({ file, signature })),
+    files: reportRuntimeChecks(cwd, signed),
   });
```
[the remark this answers is not in the document]
Comment (thecodedrift) [reply packages/cli/src/commands/check.ts]: Fixed — `signRuntimeChecks` is now per-rule resilient (returns `unreadable` instead of throwing) and a missing/unreadable `check.ts` is reported as skipped, so a malformed runtime rule no longer aborts the whole check.

*— AI Coding Agent*
